refactor: Only publish single aws-pdk package

agdimech opened the pull request:
### Changes

This PR restructures the PDK as follows:

- Individual packages are now namespaced as `@aws-pdk` and are **NOT** published to npm/pypi/maven.
- A new `aws-pdk` package is created which creates a single distributable which **IS** published to npm/pypi/maven.
- Dependencies between packages is supported via `@aws-pdk/XXX`. When packaging into the `aws-pdk`, these imports are rewritten using a combination of [`tsconfig paths`](https://www.typescriptlang.org/tsconfig#paths) and [`tsc-alias`](https://www.npmjs.com/package/tsc-alias).
- PDKProject has been refactored to pull out PDKRelease as this should now only happen in `AwsPdkProject`.
- Package tasks have been removed from individual packages as this now should only occur in `aws-pdk`.
-`aws_arch` and `cdk_graph_plugin_diagram` have been refactored slightly to ensure exported types make `JSII` happy.
- Overall DAG has been simplified and improvements to build times have been made ~8 mins
- `docs` has been pulled up to the root level
- `private` has been replaces by `projenrc`
- `PDKPipelineXXProject` has been removed from the `pipeline` package.
- NXMonorepo ensure NodeResolution is `nodenext` for TS projects
- Changed transient deps script to use shallowest transient dep version instead of greatest given we are now bundling everything together in one.
- Interfaces no longer expose required props when they aren't actually required.
- All TSProjects under a monorepo use tsconfig module reoslution settings of `nodenext` and target `es2021`
- Added a global `pdk` command which aliases underlying package manager command 🕺 
- Added an API Explorer powered by SwaggerUI to the CloudscapeReactTsWebsite which allows you to make authenticated API requests
- Added an Infra package which allows you to create boilerplate CDK code for an API + Website. This means you can get a fully functioning website + API, with all Infra in < 70 LoC 👏 

## BREAKING CHANGES

- All imports will need to be changed to `aws-pdk/<pkg>`
- All dependencies on `@aws-prototyping-sdk/*` will need to be changed to a single dependency on `aws-pdk`
- Refactored CDKGraph & CdkGraphDiagramPlugin to poly friendly definitions which will break existing consumers in TS (Py and Java never worked).
- Removed PDKPipeline projen constructs.
- Renamed `nx-monorepo` to `monorepo`
- TypeSafeApi Python runtime moved to new `python-nextgen` OpenAPI generator, which means:
  - Properties of models can be referenced as attributes rather than dictionary syntax (eg `.my_property` rather than `["my_property"]`).
  - Models are now serialised and deserialised using the `.from_json` and `.to_json` methods.
  - `request_parameters` and `request_array_parameters` are also models.
- TypeSafeApi Python handlers import location changed from `<package>.apis.tags.default_api_operation_config` to `<package>.api.operation_config`.
- TypeSafeApi Java classes for handlers/interceptors in runtime are no longer static subclasses of the `Handlers` class. The import location has changed from eg `<package>.api.Handlers.SayHello` to `<package>.api.handlers.say_hello.SayHello`.

### Remaining items
- [ ] Update Github workflow and re-enable doc publishing once written
- [ ] Update documentation including migration guide
- [ ] Document experimental vs stable
- [x] Bump deps
- [x] Create Infra project that gens infra for API + Website (Blueprint)
- [x] Clean up MonorepoTs props to categorize yarn vs pnpm specific props
- [x] Resolve #365
- [x] Refactor cloudscape construct to use native AppLayout oppose to NorthStar AppLayout
- [x] Relax ESLINT/TSC errors in website for dev
- [x] Namespace scripts
- [x] Rename NXMonorepo
- [x] Add powertools to handlers
- [x] Resolve #527
- [x] Refactor projen constructs to remove unneeded props i.e. defaultReleaseBranch
- [x] Cherry pick @cogwirrel changes

agdimech committed:
refactor: add back uber package

cogwirrel commented on packages/aws-pdk/src/aws_arch/src/index.ts:
```diff
@@ -0,0 +1,5 @@
+/*! Copyright [Amazon.com](http://amazon.com/), Inc. or its affiliates. All Rights Reserved.
```
I think you want to gitignore `packages/aws-pdk/src` :)

agdimech replied:
Good catch - updated!

cogwirrel reviewed (commented):
Looking great so far!

You'll also need to update the import in the `addDeps` for the tsapi Java/Python infra projects :)

agdimech committed:
Merge branch 'mainline' into refactor/uber

agdimech committed:
fix: TS WORKS

agdimech committed:
fix: python support

agdimech committed:
fix: all lang support

agdimech committed:
fix: resolve jest issue in graph diagram plugin

agdimech committed:
fix: fix typo with gen script

agdimech committed:
fix: refactor interfaces to remove unused properties

agdimech committed:
fix: do not make projen generated interfaces lintable

agdimech committed:
refactor(cdk-graph): refactor to ensure filters work in all langs

JeremyJonas reviewed (approved):
Awesome! Great fix :D

agdimech committed:
refactor(cdk-graph-plugin-diagram): fix broken tests

agdimech committed:
refactor(nx-monorepo): rename to monorepo

agdimech committed:
refactor: namespace scripts by package name

agdimech committed:
Merge branch 'mainline' into refactor/uber

agdimech committed:
fix(cloudscape-react-ts-website): revert to cloudscape native AppLayout

agdimech committed:
fix: update cloudscape snapshot

cogwirrel committed:
feat(type-safe-api): upgrade to python nextgen generator for python runtime to improve ux

Note that this is a breaking change since it changes import paths and the generated models. This
generator is more robust and allows for dot notation access of properties which is a little more
type safe!

agdimech committed:
fix: convert link-bundled-transitive-deps to js

agdimech committed:
fix: use nodenext for ts subprojects

agdimech committed:
fix: resolve issue in transitive dep script

agdimech committed:
fix: add missing snap

agdimech committed:
feat: add global pdk script

agdimech committed:
feat: add global pdk exec command

cogwirrel committed:
feat(type-safe-api): refactor java runtime into separate file per class

Due to a limitation in OpenAPI generator I previously generated all of the handler wrappers etc as
static subclasses of a Handlers class, since I didn't have enough control to generate a file per
desired class. With the post processing built for the handler projects, we can now split into
separate files which makes autocomplete for java a bit more intuitive.

agdimech committed:
feat: add API Explorer to Cloudscape website

cogwirrel committed:
feat(type-safe-api): generate lambda functions to reference generated handlers

Generate a lambda function construct for each operation annotated with the @handler trait as part of
the generated infrastructure package to simplify referencing the built artifacts for the handlers.

agdimech committed:
feat: add infrastructure package with ts implementation

agdimech committed:
fix: update snaps

agdimech committed:
feat: add python infra support

agdimech committed:
fix: update github release workflow

agdimech committed:
Merge branch 'mainline' into refactor/uber

agdimech committed:
feat: add Java Infra support

agdimech committed:
fix: resolve issue with java snaps

agdimech committed:
fix: change to json serializer

agdimech committed:
fix: bump deps and resolve dependabot issues

agdimech committed:
fix: resolve issues with diagram gen

agdimech committed:
fix: simplify release workflow
